UPGRADING.md

```diff
@@ -0,0 +1,372 @@
+    }
+}
+```
+
+Here's the same HTTP server written for `v2`. Please note that only header based authentication with a bearer token is
+supported by the `WithHeaderAuthorization` middleware. Cookie based authentication is not supported.
+
+```go
+import (
+    "net/http"
+
+    "github.com/clerk/clerk-sdk-go/v2"
+    clerkhttp "github.com/clerk/clerk-sdk-go/v2/http"
+)
+
+func main() {
+    clerk.SetKey("sk_live_XXX")
+    mux := http.NewServeMux()
+    mux.Handle("/session", clerkhttp.WithHeaderAuthorization()(http.HandlerFunc(handleSession)))
+    http.ListenAndServe(":3000", mux)
+}
+
+func handleSession(w http.ResponseWriter, r *http.Request) {
+    sessionClaims, ok := clerk.SessionClaimsFromContext(r.Context())
+    if ok {
+        // claims contain session information
+    } else {
+        // there is no active session (non-authenticated user)
+    }
+}
+```
+
+If you're dealing with multiple Clerk API keys, you can pass a `jwks.Client` to the `WithHeaderAuthorization` middleware.
+
+```go
+import (
+    "net/http"
+
+    "github.com/clerk/clerk-sdk-go/v2"
+    clerkhttp "github.com/clerk/clerk-sdk-go/v2/http"
+    "github.com/clerk/clerk-sdk-go/v2/jwks"
+)
+
+func main() {
+    config := &jwks.ClientConfig{}
+    config.Key = "sk_live_XXX"
+    client := jwks.NewClient(config)
+    mux := http.NewServeMux()
+    withAuth := clerkhttp.WithHeaderAuthorization(
+        clerkhttp.JWKSClient(client),
+    )
+    mux.Handle("/session", withAuth(http.HandlerFunc(handleSession)))
+    http.ListenAndServe(":3000", mux)
+}
+
+func handleSession(w http.ResponseWriter, r *http.Request) {
+    sessionClaims, ok := clerk.SessionClaimsFromContext(r.Context())
+    if ok {
+        // claims contain session information
+    } else {
+        // there is no active session (non-authenticated user)
+    }
+}
+```
+
+### Available middleware options
+
+All available middleware options are preserved in the `v2` version of the library, but they have been renamed.
+
+Name in v1 | Name in v2
+-----------|------------
+`WithAuthorizedParty` | `AuthorizedParty` and `AuthorizedPartyMatches`
+`WithLeeway` | `Leeway`
+`WithJWTVerificationKey` | `JSONWebKey`
+`WithSatelliteDomain` | `Satellite`
+`WithProxyURL` | `ProxyURL`
+`WithCustomClaims` | `CustomClaimsConstructor`
+n/a | `JWKSClient`
+
+## Verify tokens
+
+The `clerk.VerifyToken` method in version `v1` of the Clerk Go SDK has been renamed to `jwt.Verify` in `v2`.
+
+The method accepts the same parameters, with two important differences.
+
+- The JSON web key with which the token will be verified is a required parameter.
+- The method will not cache the JSON web key.
+
+In the `v1` version, the `clerk.VerifyToken` method would trigger an HTTP request to the Clerk Backend API to
+fetch the JSON web key and would cache its value for one hour.
+
+However, fetching the key and caching it should not be the responsibility of the verification process so low
```

Comment:
❓ Not sure if the motivation is necessary, but I believe it may be welcomed. You can consider linking an example to how the middleware performs the caching process so that:
- There's no misunderstanding that the middleware moves away from caching.
- Someone can see an actual real-world example.